baseclasses/decorators.py

```diff
@@ -0,0 +1,26 @@
+import functools
+import unittest
+
+
+def require_mpi(func):
+    try:
+        from mpi4py import MPI  # noqa
+
+        # this is a wrapper on func
+        # but we inject the MPI option in the signature
+        def wrapper(*args, **kwargs):
+            func(*args, **kwargs)
+
+        return wrapper
+    except ImportError:
+        msg = "mpi4py is not installed."
+        if not isinstance(func, type):
+            # this wraps obj, which is the actual function
+            @functools.wraps(func)
```

Comment:
Maybe we can use `from functools import wraps` to be cleaner?

Reply:
I think in general for decorators people tend to keep the full path, not sure why but this is what I have been seeing in other codes.